Assets/MixedRealityToolkit.SDK/Inspectors/Experimental/UX/ObjectManipulator/ObjectManipulatorInspector.cs

```diff
@@ -119,6 +125,45 @@ public override void OnInspectorGUI()
             var mh = (ObjectManipulator)target;
             var rb = mh.HostTransform.GetComponent<Rigidbody>();
 
+            EditorGUILayout.Space();
+            constraintsFoldout = EditorGUILayout.Foldout(constraintsFoldout, "Constraints", true);
+
+            if (constraintsFoldout)
+            {
+                if (EditorGUILayout.DropdownButton(new GUIContent("Add Constraint"), FocusType.Keyboard))
+                {
+                    // create the menu and add items to it
+                    GenericMenu menu = new GenericMenu();
+
+                    var type = typeof(TransformConstraint);
+                    var types = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
I feeeeeeeeeeeeel like we've seen some issues with GetAssemblies before in a couple of locations:

1) There are some assemblies that we've seen exceptions thrown when GetTypes() get called
2) I think there were some issues with .NET backend (i.e. as long as this is getting compiled away during that, we should be good).

We have some cases like this:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/2c711ab4d9f059fa0b12761a10188a051c7785b4/Assets/MixedRealityToolkit/Extensions/AssemblyExtensions.cs

I should probably write some sort of script that checks for usage of this going forward.

Reply:
> I should probably write some sort of script that checks for usage of this going forward.
These have been very useful in the past :-)

Reply:
Should I use `AssemblyExtensions` instead? To be honest, I wasn't entirely happy with this as a solution to discoverability. Constraints are certainly easier to find and reason about, but it feels a bit clunky using reflection and the highlight system (which sometimes does some weird stuff). Again, open to suggestions.

Reply:
Just tried `AssetDatabase.FindAssets` now. I can use it to get the `TransformConstraint` script asset, but it doesn't get derived types, so I'm not sure we can use it.